api/src/helpers/transferCredits.ts

```diff
@@ -324,33 +314,30 @@ const mergeDuplicateCourseResults = (toInsertCourse: TransferredCourseRow[]) =>
 };
 
 /** Organize the data in the database */
-export const organizeLegacyTransfers = async (rows: TransferData[]) => {
+export const organizeLegacyTransfers = async (rows: ExtendedTransferData[]) => {
   const allAps = await getAPIApExams();
-  // console.log('DEBUG: ALL APS: ');
-  const lookup: Record<string, Omit<TransferredMiscSelectedRow, 'courseName'>> = {};
+
+  // Determine the unique transfers
+  const lookup: Record<string, Omit<GroupedUncategorizedTransfer, 'courseName'>> = {};
 
   rows.forEach((row) => {
     const existing = lookup[row.name];
     if (existing) {
       existing.count++;
       existing.totalUnits += row.units ?? 0;
     } else {
-      lookup[row.name] = { totalUnits: row.units ?? 0, count: 1 };
+      lookup[row.name] = { totalUnits: row.units ?? 0, count: 1, score: row.score ?? undefined };
     }
   });
-
-  const uniqueRows: TransferredMiscSelectedRow[] = Object.entries(lookup).map(([courseName, data]) => ({
+  const uniqueRows: GroupedUncategorizedTransfer[] = Object.entries(lookup).map(([courseName, data]) => ({
     courseName,
     ...data,
   }));
 
-  // for (const ap of allAps) {
-  //   console.log(`  - '${ap.fullName}' (cat: '${ap.catalogueName}')`);
-  // }
-  // Obtain all relevant transfers (invalid ones that cannot be categorized are filtered out here)
-  const transfers: TransferredMiscSelectedRow[] = uniqueRows
-    //.limit(40) // For testing, only look at a few entries
-    .filter((transfer) => transfer.courseName != null && transfer.count != 0) as TransferredMiscSelectedRow[];
+  // Obtain/clean up all relevant transfers (invalid ones that cannot be categorized are filtered out here)
+  const transfers: GroupedUncategorizedTransfer[] = uniqueRows
+    .filter((transfer) => transfer.courseName != null && transfer.count != 0)
+    .map((transfer) => ({ ...transfer, courseName: transfer.courseName.trim() })) as GroupedUncategorizedTransfer[];
 
   // Get all the users who explicitly have an AB subscore
   let hasSubscore = false;
```

Comment:
Even the for loop here can be a helper function

something like
`const hasABSubscore = checkABSubscore(transfers, allAps)`